File: src/main/java/org/codelibs/fess/ds/salesforce/api/sobject/SAccount.java
/*
 * Copyright 2012-2022 CodeLibs Project and the Others.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */
package org.codelibs.fess.ds.salesforce.api.sobject;

import java.util.Arrays;
import java.util.List;

import org.codelibs.fess.ds.salesforce.api.SearchLayout;

/** 取引先 */
public class SAccount extends SearchLayout {

    /** 取引先 ID */
    protected static final String ID = "Id";
    /** 取引先名 */
    protected static final String NAME = "Name";
    /** 取引先 種別 */
    protected static final String TYPE = "Type";
    /** 町名・番地(請求先) */
    protected static final String BILLING_STREET = "BillingStreet";
    /** 市区郡(請求先) */
    protected static final String BILLING_CITY = "BillingCity";
    /** 都道府県(請求先) */
    protected static final String BILLING_STATE = "BillingState";
    /** 郵便番号(請求先) */
    protected static final String BILLING_POSTAL_CODE = "BillingPostalCode";
    /** 国(請求先) */
    protected static final String BILLING_COUNTRY = "BillingCountry";
    /** Billing Geocode Accuracy */
    protected static final String BILLING_GEOCODE_ACCURACY = "BillingGeocodeAccuracy";
    /** 町名・番地(納入先) */
    protected static final String SHIPPING_STREET = "ShippingStreet";
    /** 市区郡(納入先) */
    protected static final String SHIPPING_CITY = "ShippingCity";
    /** 都道府県(納入先) */
    protected static final String SHIPPING_STATE = "ShippingState";
    /** 郵便番号(納入先) */
    protected static final String SHIPPING_POSTAL_CODE = "ShippingPostalCode";
    /** 国(納入先) */
    protected static final String SHIPPING_COUNTRY = "ShippingCountry";
    /** Shipping Geocode Accuracy */
    protected static final String SHIPPING_GEOCODE_ACCURACY = "ShippingGeocodeAccuracy";
    /** 取引先 電話 */
    protected static final String PHONE = "Phone";
    /** 取引先 Fax */
    protected static final String FAX = "Fax";
    /** 取引先番号 */
    protected static final String ACCOUNT_NUMBER = "AccountNumber";
    /** Web サイト */
    protected static final String WEBSITE = "Website";
    /** 写真の URL */
    protected static final String PHOTO_URL = "PhotoUrl";
    /** 産業コード */
    protected static final String SIC = "Sic";
    /** 業種 */
    protected static final String INDUSTRY = "Industry";
    /** 会社形態 */
    protected static final String OWNERSHIP = "Ownership";
    /** 株式コード */
    protected static final String TICKER_SYMBOL = "TickerSymbol";
    /** 取引先 説明 */
    protected static final String DESCRIPTION = "Description";
    /** 取引先 評価 */
    protected static final String RATING = "Rating";
    /** 取引先 部門 */
    protected static final String SITE = "Site";
    /** 作成日 */
    protected static final String CREATED_DATE = "CreatedDate";
    /** 最終更新日 */
    protected static final String LAST_MODIFIED_DATE = "LastModifiedDate";
    /** System Modstamp */
    protected static final String SYSTEM_MODSTAMP = "SystemModstamp";
    /** 最終活動日 */
    protected static final String LAST_ACTIVITY_DATE = "LastActivityDate";
    /** 最終閲覧日 */
    protected static final String LAST_VIEWED_DATE = "LastViewedDate";
    /** 最終参照日 */
    protected static final String LAST_REFERENCED_DATE = "LastReferencedDate";
    /** Data.com キー */
    protected static final String JIGSAW = "Jigsaw";
    /** Jigsaw Company ID */
    protected static final String JIGSAW_COMPANY_ID = "JigsawCompanyId";
    /** 状況をクリーンアップ */
    protected static final String CLEAN_STATUS = "CleanStatus";
    /** 取引先ソース */
    protected static final String ACCOUNT_SOURCE = "AccountSource";
    /** D-U-N-S 番号 */
    protected static final String DUNS_NUMBER = "DunsNumber";
    /** 取引形態 */
    protected static final String TRADESTYLE = "Tradestyle";
    /** NAICS コード */
    protected static final String NAICS_CODE = "NaicsCode";
    /** NAICS の説明 */
    protected static final String NAICS_DESC = "NaicsDesc";
    /** 開始年 */
    protected static final String YEAR_STARTED = "YearStarted";
    /** 産業区分の説明 */
    protected static final String SIC_DESC = "SicDesc";

    protected static final List<String> contents =
            Arrays.asList(NAME, TYPE, PHONE, FAX, WEBSITE, TICKER_SYMBOL, DESCRIPTION, SITE, DUNS_NUMBER);

    @Override
    public String getTitle() {
        return NAME;
    }

    @Override
    public List<String> getContents() {
        return contents;
    }

}

File: src/main/kotlin/org/codelibs/fess/ds/salesforce/api/sobject/SAccount.kt
/*
 * Copyright 2012-2019 CodeLibs Project and the Others.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */
package org.codelibs.fess.ds.salesforce.api.sobject

import java.util.*

/** 取引先 */
data class SAccount(
        /** 取引先 ID */
        override val id: String,
        /** 取引先名 */
        val name: String,
        /** 取引先 種別 */
        val type: String?,
        /** 町名・番地(請求先) */
        val billingStreet: String?,
        /** 市区郡(請求先) */
        val billingCity: String?,
        /** 都道府県(請求先) */
        val billingState: String?,
        /** 郵便番号(請求先) */
        val billingPostalCode: String?,
        /** 国(請求先) */
        val billingCountry: String?,
        /** Billing Geocode Accuracy */
        val billingGeocodeAccuracy: String?,
        /** 町名・番地(納入先) */
        val shippingStreet: String?,
        /** 市区郡(納入先) */
        val shippingCity: String?,
        /** 都道府県(納入先) */
        val shippingState: String?,
        /** 郵便番号(納入先) */
        val shippingPostalCode: String?,
        /** 国(納入先) */
        val shippingCountry: String?,
        /** Shipping Geocode Accuracy */
        val shippingGeocodeAccuracy: String?,
        /** 取引先 電話 */
        val phone: String?,
        /** 取引先 Fax */
        val fax: String?,
        /** 取引先番号 */
        val accountNumber: String?,
        /** Web サイト */
        val website: String?,
        /** 写真の URL */
        val photoUrl: String?,
        /** 産業コード */
        val sic: String?,
        /** 業種 */
        val industry: String?,
        /** 会社形態 */
        val ownership: String?,
        /** 株式コード */
        val tickerSymbol: String?,
        /** 取引先 説明 */
        val description: String?,
        /** 取引先 評価 */
        val rating: String?,
        /** 取引先 部門 */
        val site: String?,
        /** 作成日 */
        override val createdDate: Date,
        /** 最終更新日 */
        override val lastModifiedDate: Date,
        /** System Modstamp */
        val systemModstamp: Date,
        /** 最終活動日 */
        val lastActivityDate: Date?,
        /** 最終閲覧日 */
        val lastViewedDate: Date?,
        /** 最終参照日 */
        val lastReferencedDate: Date?,
        /** Data.com キー */
        val jigsaw: String?,
        /** Jigsaw Company ID */
        val jigsawCompanyId: String?,
        /** 状況をクリーンアップ */
        val cleanStatus: String?,
        /** 取引先ソース */
        val accountSource: String?,
        /** D-U-N-S 番号 */
        val dunsNumber: String?,
        /** 取引形態 */
        val tradestyle: String?,
        /** NAICS コード */
        val naicsCode: String?,
        /** NAICS の説明 */
        val naicsDesc: String?,
        /** 開始年 */
        val yearStarted: String?,
        /** 産業区分の説明 */
        val sicDesc: String?
) : SObject {
    override fun title(): String = "${super.title()}"
    override fun content(): String = listOfNotNull("").joinToString("\n")
    override val objectType: SObjects = SObjects.Account
}
